cups/file.c

```diff
@@ -704,6 +704,171 @@ cupsFileGetChar(cups_file_t *fp)	/* I - CUPS file */
   return (*(fp->ptr)++ & 255);
 }
 
+/*
+ * 'cupsFileGetConfAndComments()' - Get a line and Comments from a configuration file.
+ */
+
+char *					/* O  - Line read or @code NULL@ on end of file or error */
+_cupsFileGetConfAndComments(
+    cups_file_t *fp,	    /* I  - CUPS file */
+    char        *buf,	    /* O  - String buffer */
+    size_t      buflen,	    /* I  - Size of string buffer */
+    char        **value,    /* O  - Pointer to value */
+    int         *linenum,   /* IO - Current line number */
+    char        *comment,   /* I  - Comment */
+    int      *commentlen, /* I  - Size of comment buffer */
+    int         *comment_linenum)  /* IO - Comment line number */
+{
+  char	*ptr;				/* Pointer into line */
+
+
+ /*
+  * Range check input...
+  */
+
+  DEBUG_printf(("2cupsFileGetConfAndComments(fp=%p, buf=%p, buflen=" CUPS_LLFMT
+                ", value=%p, linenum=%p, comment=%p, commentlen=" CUPS_LLFMT ", comment_linenum=%p)", (void *)fp, (void *)buf, CUPS_LLCAST buflen, (void *)value, (void *)linenum, (void *)comment, CUPS_LLCAST commentlen, (void *)comment_linenum));
+
+  if (!fp || (fp->mode != 'r' && fp->mode != 's') ||
+      !buf || buflen < 2 || !value || !comment || *commentlen<2)
+  {
+    if (value)
+      *value = NULL;
+
+    return (NULL);
+  }
+
+ /*
+  * Read the next non-comment line...
+  */
+
+  *value = NULL;
+
+  while (cupsFileGets(fp, buf, buflen))
+  {
+    (*linenum) ++;
+    
+   /*
+    * Strip any comments...
+    */
+ 
+    if ((ptr = strchr(buf, '#')) != NULL)
+    {
+
+      if (ptr > buf && ptr[-1] == '\\')
+      {
+        // Unquote the #...
+	_cups_strcpy(ptr - 1, ptr);
+      }
+
+      else
+      {
+        // Capturing multi line comment...
+    if(*linenum == (*comment_linenum)+1)
+    {
+        // Increase the size of comment buffer when it overflows
+      if(strlen(comment) >= (*commentlen))
+      {
+        comment = realloc(comment, (*commentlen)+1024);
+      }
+      (*commentlen) += 1024;
+      strlcat(comment, "\n", *commentlen);
+      strlcat(comment, ptr, *commentlen);
+      *comment_linenum = *linenum;
+    }
+
+    	// Capturing single line comment...
+    else
+    {
+       (*commentlen) = strlen(ptr+2);
```

Comment:
This will overwrite the current size of the allocated buffer with the length of current comment, which you probably don't want to (since you want to use the value for checking whether you can write into it).